src/config/mongoose_config_parser_toml.erl

```diff
@@ -864,15 +864,17 @@ module_opt([<<"max_http_connections">>, <<"mod_push_service_mongoosepush">>|_],
     [{max_http_connections, V}];
 module_opt([<<"access">>, <<"mod_register">>|_], V) ->
     [{access, b2a(V)}];
-module_opt([<<"welcome_message">>, <<"mod_register">>|_], V) ->
-    [{welcome_message, {b2l(V)}}];
 module_opt([<<"registration_watchers">>, <<"mod_register">>|_] = Path, V) ->
     [{registration_watchers, parse_list(Path, V)}];
 module_opt([<<"password_strength">>, <<"mod_register">>|_], V) ->
     [{password_strength, V}];
 module_opt([<<"ip_access">>, <<"mod_register">>|_] = Path, V) ->
     Rules = parse_list(Path, V),
     [{ip_access, Rules}];
+module_opt([<<"welcome_message">>, <<"mod_register">>|_], V) ->
+    Subject = maps:get(<<"subject">>, V, <<>>),
+    Body = maps:get(<<"body">>, V, <<>>),
+    [{welcome_message, {binary_to_list(Subject), binary_to_list(Body)}}];
```

Comment:
I think we should call `parse_section` here for the `subject` and `body` opts

Reply:
ok, done